### PR TITLE
Dragging line from selected shape

### PR DESCRIPTION
**Pull request by PallaviMP34**

### CODE REVIEW

**Comment by PallaviMP34** [comment]
Canvas unit test

File: diagramming-tool/package.json [hunk 1]
```diff
@@ -28,7 +26,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom --testMatch='**/App.test.js'",
+    "test": "react-scripts test --env=jsdom --testMatch=**/Component/Dashboard/Canvas.test.jsx --coverage",
```
Comment (AnushiyaM11) [inline diagramming-tool/package.json]: Just mention the general test command instead of particular file command

Comment (AnushiyaM11) [reply diagramming-tool/package.json]: You can change this now or in the next PR just let me know... If in next PR i can approve this

**Review by AnushiyaM11 — APPROVED** [review]

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx [hunk 2]
```diff
@@ -6,13 +6,19 @@ import { MdDeleteForever, MdFileOpen } from "react-icons/md";
 import { TfiSave } from "react-icons/tfi";
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/Dashboard/Canvas.jsx:6]: unused import

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx [hunk 3]
```diff
@@ -6,13 +6,19 @@ import { MdDeleteForever, MdFileOpen } from "react-icons/md";
...
 import ShapeTypes from "./ShapeTypes";
 
+
+
 const CanvasComponent = () => {
   const [selectedShapeId, setSelectedShapeId] = useState(null);
   const [selectedShape, setSelectedShape] = useState(null);
   const [selectedButton, setSelectedButton] = useState(null);
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/Dashboard/Canvas.jsx:14]: Unused states

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx [hunk 4]
```diff
@@ -6,13 +6,19 @@ import { MdDeleteForever, MdFileOpen } from "react-icons/md";
...
   const [hoveredButton, setHoveredButton] = useState("");
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/Dashboard/Canvas.jsx:15]: unused state

**Review by Sharz-dev — CHANGES_REQUESTED** [review]

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx (hunk 2 above)
Comment (PallaviMP34) [reply diagramming-tool/src/Component/Dashboard/Canvas.jsx:6]: This import is using

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx (hunk 4 above)
Comment (PallaviMP34) [reply diagramming-tool/src/Component/Dashboard/Canvas.jsx]: removed

File: diagramming-tool/src/Component/Dashboard/Canvas.jsx (hunk 3 above)
Comment (PallaviMP34) [reply diagramming-tool/src/Component/Dashboard/Canvas.jsx:14]: changes made

**Review by Sharz-dev — APPROVED** [review]

**Pull request merged**